pandas/core/arrays/string_arrow.py

```diff
@@ -330,17 +330,47 @@ def _str_contains(
             result[isna(result)] = bool(na)
         return result
 
-    def _str_startswith(self, pat: str, na=None):
-        result = pc.starts_with(self._pa_array, pattern=pat)
+    def _str_startswith(self, pat: str | tuple[str, ...], na: Scalar | None = None):
+        if isinstance(pat, str):
+            result = pc.starts_with(self._pa_array, pattern=pat)
+        elif isinstance(pat, tuple) and all(isinstance(x, str) for x in pat):
```

Comment:
`pat` is checked to be either a str or tuple in caller, so no need to check it here. I would be for not checking each individual element of the tuple - if you remove this an pass e.g. `pat=("a", 5)`, what error message do you get?

Reply:
This will then yield:

``` python
>>> import pandas as pd
>>> values = ["a", None, "b"]
>>> a = pd.Series(values, dtype="string[pyarrow]")
>>> a.str.endswith(pat=("a", 5))
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
/workspaces/pandas-chronojon/scratch/error_testing.ipynb Cell 2 line 5
      3 values = ["a", None, "b"]
      4 a = pd.Series(values, dtype="string[pyarrow]")
----> 5 a.str.endswith(pat=("a", 5))

File /workspaces/pandas-chronojon/pandas/core/strings/accessor.py:136, in forbid_nonstring_types.<locals>._forbid_nonstring_types.<locals>.wrapper(self, *args, **kwargs)
    131     msg = (
    132         f"Cannot use .str.{func_name} with values of "
    133         f"inferred dtype '{self._inferred_dtype}'."
    134     )
    135     raise TypeError(msg)
--> 136 return func(self, *args, **kwargs)

File /workspaces/pandas-chronojon/pandas/core/strings/accessor.py:2527, in StringMethods.endswith(self, pat, na)
   2525     msg = f"expected a string or tuple, not {type(pat).__name__}"
   2526     raise TypeError(msg)
-> 2527 result = self._data.array._str_endswith(pat, na=na)
   2528 return self._wrap_result(result, returns_string=False)

File /workspaces/pandas-chronojon/pandas/core/arrays/string_arrow.py:372, in ArrowStringArray._str_endswith(self, pat, na)
    369         result = pc.ends_with(self._pa_array, pattern=pat[0])
    371         for p in pat[1:]:
--> 372             result = pc.or_(result, pc.ends_with(self._pa_array, pattern=p))
    373 # else:
    374 #     raise TypeError("pat must be of type str or tuple[str, ...]")
    375 if not isna(na):

File /usr/local/lib/python3.10/site-packages/pyarrow/compute.py:258, in _make_generic_wrapper.<locals>.wrapper(memory_pool, options, *args, **kwargs)
    256 else:
    257     option_args = ()
--> 258 options = _handle_options(func_name, options_class, options,
    259                           option_args, kwargs)
    260 if args and isinstance(args[0], Expression):
    261     return Expression._call(func_name, list(args), options)

File /usr/local/lib/python3.10/site-packages/pyarrow/compute.py:221, in _handle_options(name, options_class, options, args, kwargs)
    216     if options is not None:
    217         raise TypeError(
    218             "Function {!r} called with both an 'options' argument "
    219             "and additional arguments"
    220             .format(name))
--> 221     return options_class(*args, **kwargs)
    223 if options is not None:
    224     if isinstance(options, dict):

File /usr/local/lib/python3.10/site-packages/pyarrow/_compute.pyx:1089, in pyarrow._compute.MatchSubstringOptions.__init__()

File /usr/local/lib/python3.10/site-packages/pyarrow/_compute.pyx:1072, in pyarrow._compute._MatchSubstringOptions._set_options()

File stringsource:15, in string.from_py.__pyx_convert_string_from_py_std__in_string()

TypeError: expected bytes, int found
```

This is probably fine. The tuple elements could be checked at accessor level to improve the error message but I don't think, this would be too hard to understand. I will drop the tuple check and turn it into an else case.

Reply:
Fixed in [6da713](https://github.com/pandas-dev/pandas/pull/54943/commits/6da713173217e3964541f3d50075a91d85daeeae).

tuple element check could be added to `pandas/core/strings/accessor.py:2527` but is probably a bit pedantic. Pyarrow's `pc.starts_with` method  can also work with bytes and you could then circumvent the string check by wrapping it in a tuple.